opal/corelib/runtime.js

```diff
@@ -1189,19 +1189,23 @@
     var subscriber, subscribers = Opal.stub_subscribers,
         i, ilength = stubs.length,
         j, jlength = subscribers.length,
-        method_name, stub;
+        method_name, stub,
+        opal_stubs = Opal.stubs;
 
     for (i = 0; i < ilength; i++) {
       method_name = stubs[i];
-      // Save method name to populate other subscribers with this stub
-      Opal.stubs[method_name] = true;
-      stub = Opal.stub_for(method_name);
 
-      for (j = 0; j < jlength; j++) {
-        subscriber = subscribers[j];
+      if(!opal_stubs.hasOwnProperty(method_name)) {
+        // Save method name to populate other subscribers with this stub
+        opal_stubs[method_name] = true;
+        stub = Opal.stub_for(method_name);
 
-        if (!(method_name in subscriber)) {
-          subscriber[method_name] = stub;
+        for (j = 0; j < jlength; j++) {
```

Comment:
I tried a bunch of things to make this fail and I couldn't, but you're probably a lot more familiar with the bootstrapping process than I am. Do you know if this scenario fails with this PR?

The flow, as I understand it, would be something like, given the following file:

```ruby
require 'opal-jquery'

# I don't actually know if this API is right anymore, it's been a while since I used opal-jquery.
Element['#element'].not_actually_a_method!
```

When Opal runs `Opal.add_stubs(['not_actually_a_method!'])`, `OpalJQuery` will already be registered as a subscriber by the `require` line at the top, right?